openedx_lti_tool_plugin/deep_linking/forms.py

```diff
@@ -0,0 +1,88 @@
+"""Django Forms."""
+from typing import List, Optional, Set, Tuple
+
+from django import forms
+from django.http.request import HttpRequest
+from django.urls import reverse
+from django.utils.translation import gettext as _
+from pylti1p3.deep_link_resource import DeepLinkResource
+
+from openedx_lti_tool_plugin.apps import OpenEdxLtiToolPluginConfig as app_config
+from openedx_lti_tool_plugin.edxapp_wrapper.learning_sequences import course_context
+
+
+class DeepLinkingForm(forms.Form):
+    """Deep Linking Form."""
+
+    def __init__(self, *args: tuple, request=None, **kwargs: dict):
+        """Class __init__ method.
+
+        Initialize class instance attributes and add `content_items` field.
+
+        Args:
+            *args: Variable length argument list.
+            request: HttpRequest object.
+            **kwargs: Arbitrary keyword arguments.
+
+        """
+        super().__init__(*args, **kwargs)
+        self.fields['content_items'] = forms.MultipleChoiceField(
+            choices=self.get_content_items_choices(request),
+            required=False,
+            widget=forms.CheckboxSelectMultiple,
+            label=_('Courses'),
+        )
+
+    def get_content_items_choices(self, request: HttpRequest) -> List[Tuple[str, str]]:
+        """Get `content_items` field choices.
+
+        Args:
+            request: HttpRequest object.
+
+        Returns:
+            List of tuples with choices for the `content_items` field.
+
+        """
+        return [
+            self.get_content_items_choice(course, request)
+            for course in course_context().objects.all()
```

Comment:
This query can become pretty big, do you think it is worth to implement a more robust field? One that has a search option and progressive loading? or maybe only retrieve the courses which are specified in a site configuration.

Reply:
@alexjmpb Right now there is this other PR that will separate the queryset into another method of this form: https://github.com/Pearson-Advance/openedx-lti-tool-plugin/pull/36

I do think we need to come up with a more robust way to handle this form since there seems to be the future requirement to filter per institutions, licenses or even select unit or components, I'm thinking on using a more dynamic form using a JSONField and a custom form template, I'm still not sure of what will be the UI requirements so for now I think is OK to handle it has we are doing on https://github.com/Pearson-Advance/openedx-lti-tool-plugin/pull/36.

What do you think?, do you have any resources that could be useful to create a more dynamic form?.